### PR TITLE
Add server support for handling requests that use codepoint remapping.

### PR DESCRIPTION
**Pull request by garretrieger**

### CODE REVIEW

File: patch_subset/codepoint_map.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include <unordered_map>
+#include <vector>
+
+#include "common/status.h"
+#include "hb.h"
+#include "patch_subset/patch_subset.pb.h"
+
+namespace patch_subset {
+
+class CodepointMap {
```
Comment (rsheeter) [inline patch_subset/codepoint_map.h:25]: describing the data this manages, what encode/decode means, and how the proto representation works would help. It's comprehensible but I felt like I spent more time than I should have looking up things like Encode ... what's Encode? Oh it's Remap... what's Remap?

File: patch_subset/codepoint_map.cc [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  // TODO(garretrieger): implement me!
+  LOG(WARNING) << "Not implemeneted yet.";
+  return StatusCode::kUnimplemented;
+}
+
+StatusCode CodepointMap::ToProto(CodepointRemappingProto* proto) const {
+  CompressedListProto* codepoint_ordering = proto->mutable_codepoint_ordering();
+
+  int last_cp = 0;
+  for (unsigned int i = 0; i < encode_map.size(); i++) {
```
Comment (rsheeter) [inline patch_subset/codepoint_map.cc:37]: considering you immediately assign i to an hb_codepoint_t can you just loop on it directly? - for (hb_codepoint_t cp_for_index = 0;...)

**Review by rsheeter — APPROVED** [review]
Needs comments for high level functionality and why it does what it does but other than that seems fine

File: patch_subset/codepoint_map.cc (hunk 2 above)
Comment (garretrieger) [reply patch_subset/codepoint_map.cc:37]: No because cp_for_index is modified by the Decode call.

**Pull request merged**